### ISSUE TITLE
Collect unused render targets before asking the embedder to create updated targets.

### ISSUE THREAD

**Issue by chinmaygarde** [flutter/flutter#51229]
This will reduce peak memory usage. This optimization must be made when an internal embedder migrates to be resilient to this optimization.

Pessimization added in https://github.com/flutter/engine/pull/16711 for b/146142979.

**Comment by pedromassangocode** [comment]
Is there a way to verify that this was  not fixed yet?

**Comment by chinmaygarde** [comment]
Yes. This is still an issue. This is a performance optimization that we should enable to reduce churn in render targets that we request the embedder to allocate for the engine.

**Comment by ardera** [comment]
@chinmaygarde quick ping, has that embedder migrated yet?

**Comment by chinmaygarde** [comment]
No. It was harder to achieve in the OpenGL backend. Perhaps we should provide guidance that the embedder use a pool of textures if frequent allocations/deallocations are a concern.

**Comment by ardera** [comment]
@chinmaygarde maybe a `collect_and_create_backing_stores` callback for the embedder compositor could work? For example, in my scenario I use an EGL window surface as the backing store, and when flutter requests a new backing store, it's hard to tell if I can fill it with the EGL window surface (so FBO with id 0) again because it hasn't been released before.

This is not a problem with backing store cache enabled or when there's only one view layer. (In the latter case you can just fill it with FBO ID 0 every time, but if there's multiple backing stores to be created you can't do that)

Definitely possible to work around but still, that'd make things easier
